### PR TITLE
[Example] Add mass_spring_3d.py (an extension library usage demo)

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
Congrats! We finally have a 3D example as many user argued.
This is [my GAMES 201 homework 1](https://forum.taichi.graphics/t/hw1-mass-spring-cloth-simulation/934).
I removed the implicit solver for simplicity reason.. but I've kept one in Taichi THREE examples.

It demos how to make use of Taichi THREE to render 3D scene in to cool images.
Also show off the Taichi GLSL usage (which is a hard dep of Taichi THREE)

video

So, would you do me a favor by uploading the image to README.md gallery? Many thanks~

### CODE REVIEW

**Comment by archibate** [comment]
Hello? I thought the visual result is really impressive. Can we replace taichi_sparse in gallery by this?

**Comment by yuanming-hu** [comment]
> Hello? I thought the visual result is really impressive. Can we replace taichi_sparse in gallery by this?

I agree this is impressive! We can actually add one more row there with some GAMES 201 examples.

Could you try using a 100x higher stiffness and 1/10 delta t? The cloth looks a bit too extensible.

**Comment by archibate** [comment]
video
Not sure if this is what you want..

File: examples/mass_spring_3d.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+F = ti.Vector(3, ti.f32, NN)
+
+
+@ti.kernel
+def init():
+    for i in ti.grouped(x):
+        x[i] = tl.vec((i + 0.5) * L - 0.5, 0.8).xzy
+
+
+links = [tl.vec(*_) for _ in [(-1, 0), (1, 0), (0, -1), (0, 1)]]
```
Comment (yuanming-hu) [inline examples/mass_spring_3d.py:39]: The new gif still doesn't look quite right, so I took a look at your code. Maybe you want
```suggestion
links = [tl.vec(*_) for _ in [(-1, 0), (1, 0), (0, -1), (0, 1), (-1, -1), (1, -1), (-1, 1), (1, 1)]]
```

Comment (yuanming-hu) [reply examples/mass_spring_3d.py:39]: (Not sure if you need to take care of the rest length of these diagonal springs.)

Comment (archibate) [reply examples/mass_spring_3d.py:39]: video
Cool! Changing the link between grids changes the behavior significantly.

**Comment by archibate** [comment]
Hello? May I merge this now? You may upload the images from GAMES 201 students later iapr.

**Pull request merged**